### DISCUSSION TITLE
isDate returns true for 'Invalid Date'

### DISCUSSION THREAD

**Discussion by knoefel** [js-fns/date-fns#2338] [Q&A] [answered]
Just noticed that `isDate` returns `true` while being used with an 'Invalid Date' (for example by passing `null` to `parseISO`).

Here is a [CodeSandbox](https://codesandbox.io/s/is-date-return-value-mtovs?file=/src/index.js) showing the behaviour.

Is this intentional and if so, why?

Thanks for clarification!

**Comment by tan75** [comment]
hi @knoefel 

thank you for raising this. 
The issue is the `isDate` implementation and it is going to be fixed shortly.

**Reply by kossnocorp** [reply]
Please see my comment: https://github.com/date-fns/date-fns/discussions/2338#discussioncomment-550803

**Comment by kossnocorp** [comment, marked as answer]
This is correct behavior, sorry for the confusion. `isDate` returns `true` if the argument is an instance of `Date`, and `Invalid Date` in fact is. If you want to check if the date is valid use [`isValid`](https://date-fns.org/v2.19.0/docs/isValid) is simply check using `isNaN`:

```js
Number.isNaN(new Date('').getTime())
//=> true
```
